### PR TITLE
Docker

### PR DESCRIPTION
**Pull request by glorysdj**
this PR is add Dockerfile for analytics-zoo Docker image.

---
This change is [Reviewable](https://reviewable.io/reviews/intel-analytics/analytics-zoo/416)

### CODE REVIEW

File: docker/README.md [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+
+### You can also specify the ANALYTICS_ZOO_VERSION and SPARK_VERSION to build a specific Analytics-Zoo image:
+
+    sudo docker build \
+        --build-arg http_proxy=http://your-proxy-host:your-proxy-port \
+        --build-arg https_proxy=https://your-proxy-host:your-proxy-port \
+        --build-arg ANALYTICS_ZOO_VERSION=0.1.0 \
+        --build-arg SPARK_VERSION=2.2.0 \
+        --rm -t intelanalytics/analytics-zoo:0.1.0-spark-2.2.0 .
+
```
Comment (jenniew) [inline docker/README.md]: Maybe we can also add instruction of how to use pre-build docker image from docker hub.

File: docker/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+RUN git config --global http.proxy $http_proxy && \ 
+    git config --global https.proxy $https_proxy &&\
+    git clone https://github.com/intel-analytics/analytics-zoo.git
+
+ADD ./download-analytics-zoo.sh /opt/work
+ADD ./start-notebook.sh /opt/work
+RUN chmod a+x download-analytics-zoo.sh && \
+    chmod a+x start-notebook.sh && \
+    ./download-analytics-zoo.sh
+
```
Comment (jenniew) [inline docker/Dockerfile]: After down load analytics zoo, do we need to remove download-analytics-zoo.sh script?

Comment (jenniew) [reply docker/Dockerfile]: also do we need to remove analytics-zoo-SPARK_2.2-0.1.0-dist.zip and get-pip.py on docker image?

File: docker/README.md (hunk 1 above)
Comment (glorysdj) [reply docker/README.md]: Already has it in the readme
sudo docker run -it --rm -p 12345:12345 -e NotebookPort=12345 -e NotebookToken=your-token intelanalytics/analytics-zoo:0.1.0-spark-2.2.0

sudo docker run -it --rm --net=host -e NotebookPort=12345 -e NotebookToken=your-token intelanalytics/analytics-zoo:0.1.0-spark-2.2.0

File: docker/Dockerfile (hunk 2 above)
Comment (glorysdj) [reply docker/Dockerfile]: the user may need these files to download new different version of zoo, and may recover from the dist file

**Pull request merged**